### PR TITLE
feat!: refactor metrics metadata config

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
##### Description

Move the metrics metadata configuration to a template. We don't do much more here, there are more changes expected later, pulling out some of the configuration to top-level properties.

I've added two different ways of customizing the configuration directly. The contents of `metadata.metrics.config.merge` will be merged into the templated configuration, while `metadata.metrics.config.override` will replace it in total.

Migration can be found here: https://github.com/SumoLogic/sumologic-kubernetes-tools/pull/380.

---

##### Checklist

- [x] Changelog updated

### CODE REVIEW

File: deploy/docs/opentelemetry_collector.md [hunk 1]
```diff
@@ -43,8 +43,18 @@ the same way for OpenTelemetry Collector like for Fluentd.
...
+
+The `metadata.metrics.config.merge` key can be used to provide configuration that will be merged
+with the Helm Chart's default configuration. It should be noted that this field is not subject to
+normal backwards compatibility guarantees, the default configuration can change even in minor
+versions while preserving the same end-to-end behaviour. Use of this field is discouraged - ideally
+the necessary customizations should be able to be achieved without touching the otel configuration
+directly. Please open an issue if your use case requires the use of this field.
+
+The `metadata.metrics.config.override` key can be used to provide configuration that will be completely
+replace the default configuration. This is an advanced feature requiring a good understanding of both
```
Comment (astencel-sumo) [inline deploy/docs/opentelemetry_collector.md:56]: > This is an advanced feature requiring a good understanding of this chart's architecture and OpenTelemetry Collector configuration

I like this warning and I think it should be applied to both `merge` and `override` properties.

Also, shouldn't the phrase "this field is not subject to normal backwards compatibility guarantees" be applied to both `merge` and `override` properties as well?

Comment (swiatekm-sumo) [reply deploy/docs/opentelemetry_collector.md:56]: I guess that for `override`, this depends on what you put there. You don't care at all about the configuration the Chart itself creates, but you might care about things like environment variables defined for OT Pods, or somesuch.

Comment (swiatekm-sumo) [reply deploy/docs/opentelemetry_collector.md]: I moved the warning to the start of the section and added a disclaimer about `override`.

**Review by rnishtala-sumo — APPROVED** [review]
looks like all the tests are passing, after we moved the configuration from `values.yaml` to `config.yaml` for metrics metadata. LGTM!

File: deploy/helm/sumologic/README.md [hunk 2]
```diff
@@ -345,7 +345,8 @@ The following table lists the configurable parameters of the Sumo Logic chart an
...
-| `metadata.metrics.config`                                                 | Configuration for metrics otelcol. See also https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/Configuration.md.                                                                                                                                                                                                                                   | See [values.yaml]                                                                                                 |
+| `metadata.metrics.config.merge`                                           | Configuration for metrics metadata otelcol, merged with defaults. See also https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/Configuration.md.                                                                                                                                                                                                    | {}                                                                                                                |
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/README.md:348]:
```suggestion
| `metadata.metrics.config.merge`                                           | Configuration for metrics metadata otelcol, merged with defaults. See also https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/configuration.md.                                                                                                                                                                                                    | {}                                                                                                                |
```

File: deploy/helm/sumologic/README.md [hunk 3]
```diff
@@ -345,7 +345,8 @@ The following table lists the configurable parameters of the Sumo Logic chart an
...
+| `metadata.metrics.config.override`                                        | Configuration for metrics metadata otelcol, replaces defaults.See also https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/Configuration.md.                                                                                                                                                                                                        | {}                                                                                                                |
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/README.md:349]:
```suggestion
| `metadata.metrics.config.override`                                        | Configuration for metrics metadata otelcol, replaces defaults.See also https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/configuration.md.                                                                                                                                                                                                        | {}                                                                                                                |
```

File: deploy/helm/sumologic/conf/metrics/otelcol/config.yaml [hunk 4]
```diff
@@ -1 +1,346 @@
...
+  ## ref: https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/processor/sourceprocessor
+  source:
+    collector: {{ .Values.sumologic.collectorName | default .Values.sumologic.clusterName | quote }}
+service:
+  telemetry:
+    logs:
+      level: {{ .Values.metadata.metrics.logLevel }}
+  extensions:
+    - health_check
+    - file_storage
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/conf/metrics/otelcol/config.yaml:320]: What about `{{ if .Values.metadata.persistence.enabled }}` that's used in the [logs PR](https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/2626/files#diff-14625ea0fb1afc1752d344678aa17df4e9618ecf05f944046c44ded34326d474R339)?

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/conf/metrics/otelcol/config.yaml:320]: Good point, I'll add it. I submitted this PR first and took more time to think about the conditionals in #2626

**Pull request merged**